src/DocumentFormat.OpenXml/Packaging/ImagePartTypeExtensions.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+
+
+namespace DocumentFormat.OpenXml.Packaging
+{
+    /// <summary>
+    /// Provides extensions to working with ImagePart and ImagePartType
+    /// </summary>
+    public static class ImagePartExtensions
+    {
+        /// <summary>
+        /// Extends an overloaded AddImagePart method that
+        /// handles the process of determining the ImagePartType
+        /// for you based on a provided file extension.
+        /// </summary>
+        /// <param name="main">The source MainDocumentPart that is adding the ImagePart.</param>
+        /// <param name="extension">The file extension of the image.</param>
+        public static ImagePart AddImagePart(this MainDocumentPart main, string extension) =>
+            main.AddImagePart(GetImagePartType(extension));
+
+        /// <summary>
+        /// Given a file extension, retrieve the associated ImagePartType value.
+        /// </summary>
+        /// <param name="extension">A file extension supported by ImagePartType.</param>    
+        public static ImagePartType GetImagePartType(this string extension)
```

Comment:
can you move this to the IMagePartTypeInfo class as it would be symmetric operations? Also, I'd prefer this not be an extension method off of string

Does it need to be public?

Reply:
Moved into `ImagePartTypeInfo` and no longer an extension method. Adjusted `AddImagePart` extension accordingly. Please see 94f53b3d12843f9c014adcbdf1c25106ae73f9b3